### PR TITLE
adds href removing from svgs in fixup-svg-string

### PR DESCRIPTION
**Pull request by tomlum**
### Resolves
Removes non data hrefs from svg upload

### Test Coverage

Tests for additions to fixup-svg-string

### CODE REVIEW

File: src/fixup-svg-string.js [hunk 1]
```diff
@@ -57,5 +57,10 @@ module.exports = function (svgString) {
     // Empty script tags and javascript executing
     svgString = svgString.replace(/<script[\s\S]*>[\s\S]*<\/script>/, '<script></script>');
 
+    // Empty script tags and javascript executing
+    svgString = svgString.replace(/<script[\s\S]*>[\s\S]*<\/script>/, '<script></script>');
```
Comment (apple502j) [inline src/fixup-svg-string.js]: Is this intentional? (If we want proper escaping we should just move DOMPurify here)

**Comment by apple502j** [comment]
Overall this looks like cat-and-mouse situation - regex based filtering is very likely to miss some edge cases (e.g. a regex filters something, which in turn adds bad things). Since we use DOMPurify anyway why not move it there?

File: test/fixup-svg-string.js [hunk 2]
```diff
@@ -142,3 +142,17 @@ test('fixupSvgString shouldn\'t correct non-image tags', t => {
     t.notEqual(dontFix.indexOf('img/png'), -1);
     t.end();
 });
+
+test('fixupSvgString SHOULD correct hrefs that are NOT data links', t => {
+    const fixed = fixupSvgString('<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:foo="http://www.w3.org/1999/xlink" xml:space="preserve"><image foo:href="https://placekitten.com/64/64"/></svg>');
+
+    t.equal(fixed.indexOf('https://placekitten.com/64/64'), -1);
```
Comment (benjiwheeler) [inline test/fixup-svg-string.js:148]: I think we should check for the whole string we expect -- to make sure the regex doesn't match too much

File: test/fixup-svg-string.js [hunk 3]
```diff
@@ -142,3 +142,17 @@ test('fixupSvgString shouldn\'t correct non-image tags', t => {
     t.notEqual(dontFix.indexOf('img/png'), -1);
     t.end();
 });
+
+test('fixupSvgString SHOULD correct hrefs that are NOT data links', t => {
```
Comment (benjiwheeler) [inline test/fixup-svg-string.js:145]: Maybe also make some tests that use whitespace within tags to try to obfuscate them

File: src/fixup-svg-string.js [hunk 4]
```diff
@@ -57,5 +57,8 @@ module.exports = function (svgString) {
     // Empty script tags and javascript executing
     svgString = svgString.replace(/<script[\s\S]*>[\s\S]*<\/script>/, '<script></script>');
 
+    // Remove hrefs that aren'ts data uris
+    svgString = svgString.replace(/href="(?!data).*"/, '');
```
Comment (benjiwheeler) [inline src/fixup-svg-string.js]: Might have to use a regex more like: `href\s*=\s*"(?!data)[^"]*"` to account for extra whitespace and to keep from greedily matching until the last quotes in the whole string

File: src/fixup-svg-string.js [hunk 5]
```diff
@@ -1,8 +1,11 @@
+const DOMPurify = require('isomorphic-dompurify');
```
Comment (adroitwhiz) [inline src/fixup-svg-string.js:1]: I don't see this added anywhere in the dependencies.

Also, it looks like `isomorphic-dompurify` is a completely separate project maintained by Some Guy On GitHub(tm), so there's the potential for the DOMPurify version in `isomorphic-dompurify` to become out-of-date.

It looks like all it does is set `window` using JSDOM, which you could just do manually in the tests:
```js
const jsdom = require('jsdom');
const {JSDOM} = jsdom;
const {window} = new JSDOM('<!DOCTYPE html>');
global.window = window;
```

**Comment by benjiwheeler** [comment]
Closing per the newer https://github.com/LLK/scratch-svg-renderer/pull/336 , which also takes into account what @apple502j said here about DOMPurify being a more reliable and thorough solution, and what @adroitwhiz said here about the ease of replicating what isometric-dompurify does if we want to run DOMPurify without a browser (e.g., in tests)
